src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Xamarin.Android.Sdk.props

```diff
@@ -29,6 +29,12 @@ Copyright (C) 2011-2016 Xamarin. All rights reserved.
 		<FrameworkPathOverride>$(TargetFrameworkRootPath)MonoAndroid\v1.0</FrameworkPathOverride>
 	</PropertyGroup>
 
+	<Target Name="DesignTimeSetup">
+		<PropertyGroup>
+			<DesignTimeBuild Condition=" '$(DesignTimeBuild)' == '' And '$(BuildingInsideVisualStudio)'=='true' And '$(BuildingOutOfProcess)'=='false' ">true</DesignTimeBuild>
```

Comment:
VS 2015 etc used to set `$(DesignTimeBuild)` for all the background builds they do. This includes Intellisense etc. We used this property to detect if we needed to run or skip long standing tasks like `GetAdditionalResourcesFromAssemblies`.

With the introduction of VS2017 (or at some point) it seems that these background builds are no longer setting `$(DesignTimeBuild)` at all. It is always empty. 

They do, however set `$(BuildingInsideVisualStudio)` to true and `$(BuildingOutOfProcess)` to false. This seems to be the only consistent difference background and foreground builds. Other than the root target which is run. However we are able to know which target is being run from within msbuild itself. So we need to make use of these two properties to figure out if we are in a background build or not. 

There is no need to set this property to empty or false at this point as it is already taken care of later in the process. [1]. Now you might argue that that code should be moved into this area. But
[2] points to a problem where XS was resetting the property after it had been initially set. So I would rather not move code around at this point without a window for a good amount of regression testing. 

[1] https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L366
[2] https://github.com/xamarin/xamarin-android/commit/0ca2c216b04c67ca0b1b6cf5d34580a4b26adac2